force-app/main/default/triggers/Count.trigger

```diff
@@ -0,0 +1,70 @@
+trigger Count on Task (after insert, after update, after delete) {
+  
+    
+      Set<Id> accIds = new Set<Id>();
+        if (trigger.isInsert) {
+            for (Task task : Trigger.new){
+                if (task.whatId !=null && String.valueOf(task.whatId).startsWith('001')) {
+                    accIds.add(task.whatId);
+                }
+            }
+        } else if (trigger.isUpdate) {
+            for (Task task : Trigger.new){
+                if (task.whatId !=null && String.valueOf(task.whatId).startsWith('001')) {
+                    accIds.add(task.whatId);
+                }
+            }
+        } else if (trigger.isDelete) {
+            for (Task task : Trigger.old){
+                if (task.whatId !=null && String.valueOf(task.whatId).startsWith('001')) {
+                    accIds.add(task.whatId);
+                }
+            }
+        }
+        
+    if (accIds.size() > 0){
+    
+         Map<Id,Account> accountMap = new Map<Id,Account>();
+            
+         Id recordTypeId= Schema.SObjectType.Account.getRecordTypeInfosByDeveloperName().get('United_States').getRecordTypeId();
+         
+         List<Account> accountsList = [Select Id,Name, Open_Tasks_Call__c, Open_Tasks_Email__c, 
```

Comment:
I would have expected to see accIds used here

Reply:
This way you update all accounts instead only affected accounts

Reply:
First of all, thank you for reviewing those classes and for your advices. I will make the required changes.

Yes, you are completely right. But the problem is that if I put 'Where Id IN :accIds' I have a problem with the count in one case. This case is when the user wants to update the Task's Related Account (he creates the task on a certain account and then he changes the related account with another account). When he does that, the  count on the previous related account is not done correctly. Not putting accIds in the select is definitely not the best solution.
I need to change something in the Trigger's logic, but I don't know exactly how to do that in a simple way.

Reply:
Not putting the 'accIds' condition in the SOQL is a big mistake, because every time the trigger fires he will query all the Accounts with RecordType='United States' and Type='Prospect', and not only the accounts related to the Tasks from Trigger.new or from Trigger.old. 

If my org has many accounts with those conditions, the risk is to reach some governor limits, like the number of records retrieved by SOQL queries and the number of rows that can be processed by DML operations in a single transaction.

That can be clearly seen by looking at the logs (related to the insertion of a task on the same account):

With 'accIds' condition:
![with accIds](https://user-images.githubusercontent.com/77618420/216398622-93d9a0e5-3f68-4531-bdcd-eb2f8ca64e34.jpg)

Without 'accIds' condition:
![without accIds](https://user-images.githubusercontent.com/77618420/216398642-2e6dd6ce-0eaf-4c34-9cc5-070c188e62c0.jpg)

Reply:
>Not putting the 'accIds' condition in the SOQL is a big mistake

I tend to agree. A clean PR is our priority for now, we can discuss missing/additional features after we have done our first round of review and changes. 

Please re request a review when you are happy/done.